megamek/src/megamek/common/pathfinder/BoardEdgePathFinder.java

```diff
@@ -307,15 +517,15 @@ private boolean isLegalMove(MovePath movePath) {
         int srcHexElevation = calculateUnitElevationInHex(srcHex, entity);
         
         boolean destinationImpassable = destHex.containsTerrain(Terrains.IMPASSABLE);
-        boolean destinationHasBuilding = destinationBuilding != null;
+        boolean destinationHasBuildingOrBridge = destinationBuilding != null;
+        boolean destinationHasBridge = destinationHasBuildingOrBridge && destHex.containsTerrain(Terrains.BRIDGE_CF);
+        boolean destinationHasBuilding = destinationHasBuildingOrBridge && destHex.containsTerrain(Terrains.BLDG_CF);
         
         // if we're going to step onto a bridge that will collapse, let's not consider going there
-        boolean destinationHasWeakBridge = destinationHasBuilding && destHex.containsTerrain(Terrains.BRIDGE_CF) &&
-                destinationBuilding.getCurrentCF(dest) > entity.getWeight();
+        boolean destinationHasWeakBridge =  destinationHasBridge && destinationBuilding.getCurrentCF(dest) < entity.getWeight();
```

Comment:
You know, as often as this check gets done (I know I coded it a few times for Princess path ranking) it kind of begs for Building to have a helper method that just takes in an entity and simply returns a boolean.